### ISSUE TITLE
Deployment on different servers not working

### ISSUE THREAD

**Issue by javialon26** [strapi/strapi#19330]
### Required System information

Launched In: 166 ms
Environment: development
OS: darwin-arm64
Strapi Version: 4.19.0
Node/Yarn Version: npm/10.2.3 node/v20.10.0 darwin arm64 workspaces/false
Edition: Community
Database: sqlite

### Describe the bug

I'm following the [deployment guide](https://docs.strapi.io/dev-docs/admin-panel-customization#different-servers) for different servers (I want to deploy the admin panel - react app - to Netlify and the server to a droplet or similar). Still, the generated index.html file contains invalid paths to the generated js files.

### Steps to reproduce the behavior

1. Create a Strapi example project using the quick start guide - https://docs.strapi.io/dev-docs/quick-start
2. Configure the server.js and admin.js files as described in the different server configuration guide - https://docs.strapi.io/dev-docs/admin-panel-customization#different-servers
3. Run NODE_ENV=production npm run build
4. The generated paths to the js files contain an extra slash, generating 404 errors:

src="//runtime~main.35ff75f1.js"
src="//main.19186bea.js"

### Expected behavior

Create the index.html with the correct paths with only one slash to access the admin site using the http://yourfrontend.com URL. I think in this context the proxy recommendation not apply because this is a front-end app with the index.html file and js files, must work with any static hosting provider.

**Comment by derrickmehaffy** [comment]
We don't support relative URLs so if you are splitting the admin from the backend both the URL key in the server and admin need to be set to absolute URLs and the admin rebuilt.

**Comment by derrickmehaffy** [comment]
If it's a docs issue that needs to go to https://GitHub.com/Strapi/documentation

**Comment by javialon26** [comment]
Thanks for the answer but that also doesn't work. If I put an absolute URL in the url parameter in the admin.js file the index.html remains the same with the double slash and the file name `//runtime~main.35ff75f1.js` after running the npm run build command.

**Comment by perryzhangateosi** [comment]
@javialon26 i'm facing the exactly same problem, either absolute url or "/" resulted in "//runtime~main.3087f3fd.js", it just won't work.  this is bothering me for three days.  
I honestly thought there is sth wrong with my config or codding until i saw this issue.
I found an alt way whics is not perfect： use "/xxx" in admin.js  url parameter,  and set an alias /xxx in your web application config file (in apache:  Alias /xxx/   xxxx ),  now the separate admin panel can be visited as your wish. But when you press F5, the page came with 404.

**Comment by derrickmehaffy** [comment]
Try building that locally first to confirm if the remote build is just failing. I'll test this on Monday when I'm back in the office to confirm.

**Comment by javialon26** [comment]
@perryzhangateosi I did the same thing but with Netlify redirects. I put only the serveAdminPanel flag in false in admin.js and the absolute URL in server.js for the back-end, so the index.html has the default /admin paths. I redirected all /admin URLs to /:

```
[[redirects]]
  from = "/admin/*"
  to = "/:splat"
  status = 200
```

This is a workaround because the build script should read the URL param properly but at least it is working now (I'm facing the same 404 issue if I reload the page, that is because in Netlify I can't put redirects with regex or similar. Try in Apache to create rewrites for all /admin js files to root and everything else to /index.html like any other SPA.)

**Comment by anil1712** [comment]
I am still getting the same issue, can anyone please tell me how to resolve this?

**Comment by tirthshah574** [comment]
@anil1712  @perryzhangateosi @javialon26 @derrickmehaffy 

I found that creating the patch in this file is the only workaround.

patches/@strapi+admin+4.23.0.patch

diff --git a/node_modules/@strapi/admin/dist/_chunks/index-sNH2VWbC.js b/node_modules/@strapi/admin/dist/_chunks/index-sNH2VWbC.js
index b284917..81e03db 100644
--- a/node_modules/@strapi/admin/dist/_chunks/index-sNH2VWbC.js
+++ b/node_modules/@strapi/admin/dist/_chunks/index-sNH2VWbC.js
@@ -619,7 +619,7 @@ const createBuildContext = async ({
   const { bundler = "webpack", ...restOptions } = options;
   const buildContext = {
     appDir,
-    basePath: `${adminPath}/`,
+    basePath: `${adminPath}`,
     bundler,
     customisations,
     cwd,

**Comment by anil1712** [comment]
@tirthshah574 I am using AWS CI/CD pipeline, do you have any idea how it will work on the fly after the build command gets run? any command line related solution you can suggest?

**Comment by derrickmehaffy** [comment]
Related TID: 6923

@bolg55 and I were trying to reproduce another issue somewhat related to this for an EE customer and in doing some testing we were able to confirm at least something is in fact broken sometime after v4.14.6 when trying to use split configs.

Below is my testing setup using Nginx:

### Hosts config for testing locally

```
// path /etc/hosts
...
127.0.0.1	api.test.local
127.0.0.1	admin.test.local
...
```

### Nginx config for Strapi backend

```
// path /etc/nginx/sites-enabled/strapi-backend.conf
server {
    # Listen HTTP
    listen 80;
    server_name api.test.local;

    # Proxy Config
    location / {
        proxy_pass http://127.0.0.1:1337;
        proxy_http_version 1.1;
        proxy_set_header X-Forwarded-Host $host;
        proxy_set_header X-Forwarded-Server $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header Host $http_host;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_pass_request_headers on;
    }
}
```

### Nginx config for Strapi admin 

```
// path /etc/nginx/sites-enabled/strapi-frontend.conf
server {
    # Listen HTTP
    listen 80;
    server_name admin.test.local;

    root /var/www/html/build;
    index index.html;

    # Proxy Config
    location / {
      try_files $uri $uri/ =404;
    }
}
```

### Strapi Admin.js config file

```js
module.exports = ({ env }) => ({
  auth: {
    secret: env("ADMIN_JWT_SECRET"),
  },
  apiToken: {
    salt: env("API_TOKEN_SALT"),
  },
  transfer: {
    token: {
      salt: env("TRANSFER_TOKEN_SALT"),
    },
  },
  flags: {
    nps: env.bool("FLAG_NPS", true),
    promoteEE: env.bool("FLAG_PROMOTE_EE", true),
  },
  url: "http://admin.test.local",
  serveAdminPanel: false,
});
```

### Strapi Server.js config file

```js
module.exports = ({ env }) => ({
  host: env("HOST", "0.0.0.0"),
  port: env.int("PORT", 1337),
  app: {
    keys: env.array("APP_KEYS"),
  },
  webhooks: {
    populateRelations: env.bool("WEBHOOKS_POPULATE_RELATIONS", false),
  },
  url: "http://api.test.local",
  proxy: true,
});
```

---

When testing this on v4.14.6 this config works perfectly however after updating to the latest v4.24.4 this no longer works and the files being loaded by the admin are completely wrong:

![image](https://github.com/strapi/strapi/assets/8593673/c4ca2189-283c-4718-8c50-1ec6b98e1ce4)

**Comment by derrickmehaffy** [comment]
Any engineer that works on this reach out to me if you need help setting up a testing environment to validate the bug and/or validate a fix

**Comment by tirthshah574** [comment]
@derrickmehaffy 
It could just be the path issue. I have given the solution above for the fix. Try patching that!

**Comment by tirthshah574** [comment]
@anil1712 
It is just a normal config after you fix the issue that I mentioned using patch package.

Here is my file for your reference

> config/admin.ts
```
export default ({ env }) => ({
  auth: {
    secret: env('ADMIN_JWT_SECRET')
  },
  apiToken: {
    salt: env('API_TOKEN_SALT')
  },
  transfer: {
    token: {
      salt: env('TRANSFER_TOKEN_SALT')
    }
  },
  watchIgnoreFiles: ['**/config/sync/**', '**/logs', './logs'],
  // Note: The administration will be accessible from the root of the domain (ex: http://yourfrontend.com/)
  // http://yourbackend.com will not serve any static admin files
  ...(env('SERVE_ADMIN_PANEL') === 'false' && { serveAdminPanel: false, url: env('FRONTEND_URL') })
});
```

**Comment by inc00ming** [comment]
@tirthshah574 could you please provide the needed steps for the patch-package usage
https://github.com/strapi/strapi/issues/19330#issuecomment-2059710630

**Comment by kristianka** [comment]
Same issue. The fix seems to be to run with `NODE_ENV=development`. Hopefully this gets fixed on Strapi 5.0, not sure how good idea it is to run production with development mode 😬

My setup: 

- Dockerized NextJS
- Dockerized Strapi
- Dockerized Postgres for DB
- Dockerized NGINX for routing

![image](https://github.com/user-attachments/assets/5a89d4fa-cde7-48b1-b54b-21845a2f04d0)
![image](https://github.com/user-attachments/assets/5285e6c9-6423-4e37-901c-d307e786ee1e)

**Comment by nlundquist** [comment]
Similar issues still persist in 5.0

**Comment by CharuGoyalS** [comment]
Same issue in version 4.25.5. Any way to get this prioritized

**Comment by thomas-coldwell** [comment]
Had the same issue here and @tirthshah574 's solution solved this perfectly for me. We are hosting all the Strapi Admin files in an S3 bucket served out via Cloudfront

**Comment by ashisherc** [comment]
cant get the separate server admin build to work with v5, the dependency paths are set without `/`, hence, the admin panel can not resolve js/css files.

**Comment by brianjd** [comment]
@hanpaine Do you know which sprint this will land in? It will be great to have a timeline.

**Comment by hanpaine** [comment]
hey @brianjd #22206 is the fix to follow - we're aiming to get it merged this week and deployed next 😄

**Comment by ashisherc** [comment]
@hanpaine the PR above is only for the V4 right? the issue is present in V5 as well

**Comment by hanpaine** [comment]
hey @ashisherc that's correct - I've duped this ticket to #22284 to track more clearly for v5. The PR fix for v5 is in progress now, similarly targeted for next week 👍

**Comment by brianjd** [comment]
@hanpaine Thanks.  But I am on v5.

**Comment by hanpaine** [comment]
@brianjd the fix will come with #21914 but this is not quite ready to go yet - coming soon!

**Comment by clanaid** [comment]
> @brianjd the fix will come with #21914 but this is not quite ready to go yet - coming soon!修复将随之而来 #21914 但这还没有完全准备好 - 即将推出！

When will this fix be released? I also encountered the publicpath error path

**Comment by hanpaine** [comment]
Hi folks, fixes on v4 and v5 will be released this Wednesday 🚀  thanks for your patience, this was a tricky one to get right 🙏
